# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/ipc/jsactor/JSActor.h
# Commit: 7229926a45c2
# Full Hash: 7229926a45c2e43136c7f8f597a92776658edbd5
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-10-14 09:52:12
# Description:
#   Bug 1660539 - Make StructuredCloneData args optional in JSActor RawMessage, r=kmag
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D93122
# ==============================================================================

diff -r 1aa2dc4b280e -r 7229926a45c2 dom/ipc/jsactor/JSActor.h
--- a/dom/ipc/jsactor/JSActor.h	Tue Oct 13 16:57:32 2020 +0000
+++ b/dom/ipc/jsactor/JSActor.h	Tue Oct 13 17:43:56 2020 +0000
@@ -60,8 +60,8 @@
   // message metadata |aMetadata|. The underlying transport should call the
   // |ReceiveMessage| method on the other side asynchronously.
   virtual void SendRawMessage(const JSActorMessageMeta& aMetadata,
-                              ipc::StructuredCloneData&& aData,
-                              ipc::StructuredCloneData&& aStack,
+                              Maybe<ipc::StructuredCloneData>&& aData,
+                              Maybe<ipc::StructuredCloneData>&& aStack,
                               ErrorResult& aRv) = 0;
 
   // Check if a message is so large that IPC will probably crash if we try to
@@ -72,8 +72,8 @@
   // Helper method to send an in-process raw message.
   using OtherSideCallback = std::function<already_AddRefed<JSActorManager>()>;
   static void SendRawMessageInProcess(const JSActorMessageMeta& aMeta,
-                                      ipc::StructuredCloneData&& aData,
-                                      ipc::StructuredCloneData&& aStack,
+                                      Maybe<ipc::StructuredCloneData>&& aData,
+                                      Maybe<ipc::StructuredCloneData>&& aStack,
                                       OtherSideCallback&& aGetOtherSide);
 
   virtual ~JSActor() = default;
@@ -131,7 +131,7 @@
     ~QueryHandler() = default;
 
     void SendReply(JSContext* aCx, JSActorMessageKind aKind,
-                   ipc::StructuredCloneData&& aData);
+                   Maybe<ipc::StructuredCloneData>&& aData);
 
     RefPtr<JSActor> mActor;
     RefPtr<Promise> mPromise;